lib/amo-client.js

```diff
@@ -121,14 +121,24 @@ Client.prototype.waitForSignedAddon = function(statusUrl, opt) {
         var data = result[1];
         lastStatusResponse = data;
 
+        // TODO: remove this when the API has been fully deployed with this
+        // change: https://github.com/mozilla/olympia/pull/1041
+        var apiReportsAutoSigning = typeof data.automated_signing !== 'undefined';
+
+        var canBeAutoSigned = data.automated_signing;
         var failedValidation = !data.valid;
         // The add-on passed validation and all files have been created.
+        // There are many checks for this state because the data will be
+        // updated incrementally by the API server.
         var signedAndReady = (data.valid && data.active && data.reviewed &&
                               data.files && data.files.length > 0);
-        // The add-on is valid but requires a manual review before it can be signed.
-        var requiresManualReview = data.valid && !data.active && !data.reviewed;
+        // The add-on is valid but requires a manual review before it can
+        // be signed.
+        var requiresManualReview = (data.valid && apiReportsAutoSigning &&
```

Comment:
JavaScript doesn't actually require these parens. Same on line 133. I'm fine with them though if you prefer it.